src/libs/auth.ts

```diff
@@ -450,23 +453,28 @@ export const processUser = (user: OidcUser | null, isSignInEvent: boolean): void
   });
 };
 
-const initializeTermsOfService = (isSignedIn: boolean, state: AuthState): TermsOfServiceStatus => {
-  return {
-    userHasAcceptedLatestTos: isSignedIn ? state.termsOfService.userHasAcceptedLatestTos : undefined,
-    permitsSystemUsage: isSignedIn ? state.termsOfService.permitsSystemUsage : undefined,
-  };
-};
-
-export const initializeAuth = _.memoize(async () => {
+// this function is only called when the browser refreshes
+export const initializeAuth = _.memoize(async (): Promise<void> => {
   // Instantiate a UserManager directly to populate the logged-in user at app initialization time.
-  // All other auth usage should use the AuthContext from authStore.
+  // All other auth usage should use the AuthContext from oidcStore.
   const userManager: UserManager = new UserManager(getOidcConfig());
-  processUser(await userManager.getUser(), false);
+
+  const errorMessage = 'Failed to initialize auth.';
+  try {
+    const initialOidcUser: OidcUser | null = await userManager.getUser();
+    if (initialOidcUser !== null) {
+      doUserLoaded(initialOidcUser);
+    }
+  } catch (e) {
+    console.error('Error in contacting oidc-client');
+    throw new Error(errorMessage);
```

Comment:
It looks like `errorMessage` isn't used anywhere else, so maybe inline it here?

```suggestion
    throw new Error('Failed to initialize auth.');
```